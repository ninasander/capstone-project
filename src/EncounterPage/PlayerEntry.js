import React from 'react'
import styled from 'styled-components/macro'

export default function PlayerEntry({ player, ac }) {
  return (
    <PlayerEntryStyled>
      <h2>{player}</h2>
      <p>AC: {ac}</p>
    </PlayerEntryStyled>
  )
}

const PlayerEntryStyled = styled.div`
  margin: 20px;
  padding: 20px;
  display: grid;
  grid-template-columns: 1fr 1fr;
  background: linear-gradient(
      180deg,
      hsla(208, 40%, 28%, 1) 0%,
      hsla(208, 40%, 23%, 1) 51%,
      hsla(207, 38%, 18%, 1) 100%
    )
    fixed no-repeat;
  color: white;
  border-radius: 5px;
  h2,
  p {
    align-self: center;
    margin: 0;
  }
`
