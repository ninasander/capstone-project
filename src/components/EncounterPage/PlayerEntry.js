import React from 'react'
import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import DisplayPlayerAndEnemy from './DisplayPlayerAndEnemy'

PlayerEntry.propTypes = {
  playerName: PropTypes.string.isRequired,
  playerArmorClass: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
    .isRequired,
  playerInitiative: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
    .isRequired,
}

export default function PlayerEntry({
  playerName,
  playerArmorClass,
  playerInitiative,
  index,
  activeIndex,
}) {
  return (
    <PlayerEntryStyled active={index === activeIndex}>
      <DisplayPlayerAndEnemy
        initiative={playerInitiative}
        name={playerName}
        armorClass={playerArmorClass}
      />
    </PlayerEntryStyled>
  )
}

const PlayerEntryStyled = styled.div`
  margin: 20px;
  padding: 15px;
  display: grid;
  grid-template-columns: 1fr 3fr 1fr;
  background: linear-gradient(
      180deg,
      hsla(208, 40%, 28%, 1) 0%,
      hsla(208, 40%, 23%, 1) 51%,
      hsla(207, 38%, 18%, 1) 100%
    )
    fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: ${(props) =>
    props.active
      ? '0 0 15px var(--highlight-blue)'
      : '4px 4px 6px rgba(8, 15, 21, 0.3)'};
  border: ${(props) => props.active && '2px solid var(--highlight-blue)'};
  /* h1 {
    align-self: center;
    margin: 0;
    font-size: 1.8rem;
  }

  p {
    align-self: center;
    margin: 0;
  } */
`
// const InitiativeStyled = styled.p`
//   justify-self: center;
//   padding: 5px;
//   background-image: url(${d20Icon});
//   background-position: center;
//   background-size: 30px;
//   background-repeat: no-repeat;
// `

// const ACStyled = styled.p`
//   text-align: center;
//   padding: 10px 0;
//   background-image: url(${shieldIcon});
//   background-size: 37px;
//   background-repeat: no-repeat;
//   background-position: center;
//   p {
//     margin: 0;
//   }
// `
