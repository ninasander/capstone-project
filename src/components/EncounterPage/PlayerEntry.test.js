import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import PlayerEntry from '../EncounterPage/PlayerEntry'
import 'jest-styled-components'

describe('PlayerEntry', () => {
  const player = 'Player 1'
  const ac = 16

  it('displays the player name', () => {
    const { getByText } = render(<PlayerEntry player={player} ac={ac} />)

    expect(getByText(player)).toBeInTheDocument()
  })

  it('displays the player AC', () => {
    const acEntry = 'AC: ' + ac
    const { getByText } = render(<PlayerEntry player={player} ac={ac} />)

    expect(getByText(acEntry)).toBeInTheDocument()
  })

  it('renders correctly', () => {
    const tree = renderer.create(<PlayerEntry player={player} ac={ac} />)
    expect(tree).toMatchSnapshot()
  })
})
