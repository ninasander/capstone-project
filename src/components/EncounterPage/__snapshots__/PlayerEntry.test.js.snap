// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerEntry renders correctly 1`] = `
.c0 {
  margin: 20px;
  padding: 20px;
  display: grid;
  grid-template-columns: 1fr 3fr;
  background: linear-gradient( 180deg,hsla(208,40%,28%,1) 0%,hsla(208,40%,23%,1) 51%,hsla(207,38%,18%,1) 100% ) fixed no-repeat;
  color: white;
  border-radius: 5px;
  box-shadow: 4px 4px 6px rgba(8,15,21,0.3);
}

.c0 h1 {
  grid-row: span 2;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c0 h2 {
  margin: 0;
  font-size: 140%;
}

.c0 p {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  margin: 0;
}

<div
  className="c0"
>
  <h1>
    20
  </h1>
  <h2>
    Player 1
  </h2>
  <p>
    AC: 
    17
  </p>
</div>
`;
